src/utils/bitcoin.ts

```diff
@@ -75,3 +78,13 @@ export function btcAddressFromParams(
         ...params,
     });
 }
+
+export async function getTxProof(electrsAPI: ElectrsAPI, btcTxId?: string, merkleProof?: Bytes, rawTx?: Bytes): Promise<[Bytes, Bytes]> {
+    if (!merkleProof || !rawTx) {
+        if(!btcTxId) {
+            throw new Error("Either the `btcTxId` or both `merkleProof` and `rawTx` must be defined to execute.");
+        }
+        [merkleProof, rawTx] = await electrsAPI.getParsedExecutionParameters(btcTxId);
+    }
```

Comment:
Scrap my comment on issue, redeem and replace, I see you are handling this here.

One thing though: we can technically also fetch everything in case the user only submits the rawTx. But it's fine if we require both I guess. Technically you can generate the btc tx id from the rawTx but it requires some extra parsing.